audit/audit-gcp.sh

```diff
@@ -6,96 +6,150 @@ set -o pipefail
+            --project="${PROJECT}" \
+            --format=json \
+            > "${PROJECT}/service-accounts/${SVCACCT}/description.json"
+        gcloud \
+            iam service-accounts get-iam-policy "${SVCACCT}" \
+            --project="${PROJECT}" \
+            --format=json \
+            > "${PROJECT}/service-accounts/${SVCACCT}/iam.json"
+    done
+
+    echo "#### ${PROJECT} Roles ####"
+    gcloud \
+        iam roles list \
+        --project="${PROJECT}" \
+        --format="value(name)" \
+    | while read -r ROLE_PATH; do
+        mkdir -p "${PROJECT}/roles"
         ROLE=$(basename "${ROLE_PATH}")
-        gcloud iam roles --project="${PROJECT}" describe "${ROLE}" \
-               --format=json > "${PROJECT}/roles/${ROLE}.json"
+        gcloud \
+            iam roles describe "${ROLE}" \
+            --project="${PROJECT}" \
+            --format=json \
+            > "${PROJECT}/roles/${ROLE}.json"
     done
 
-    echo "#### Iterating over ${PROJECT} Services: ####"
-    mkdir -p "${PROJECT}/services"
-    gcloud services list --filter state:ENABLED --format="${format}" > "${PROJECT}/services/enabled.${format}"
-    for service in $(gcloud services list --filter state:ENABLED --format=json \
-                    | jq -r .[].config.name | sed s:.googleapis.com:: \
-                    | sort)
-    do
-        case "${service}" in
+    echo "#### Services ####"
+    gcloud \
+        services list \
+        --filter="state:ENABLED" \
+        --format=json \
+        > "${PROJECT}/services/enabled.json"
+    gcloud \
+        services list \
+        --filter="state:ENABLED" \
+        --format="value(config.name)" \
+    | sed 's/.googleapis.com//' \
+    | while read -r SVC; do
+        case "${SVC}" in
             compute)
-                echo "TODO: ${service} Needs compute.projects.get"
-                #### gcloud compute project-info describe
-                #### gcloud compute instances list --format="${format}" > "${PROJECT}/services/compute.instances.${format}"
-                #### gcloud compute disks list --format="${format}" > "${PROJECT}/services/compute.disks.${format}"
+                mkdir -p "${PROJECT}/services/${SVC}"
+                gcloud \
+                    compute project-info describe \
+                    --project="${PROJECT}" \
+                    --format=json \
+                    > "${PROJECT}/services/${SVC}/project-info.json"
                 ;;
             container)
-                echo "${service}"
+                mkdir -p "${PROJECT}/services/${SVC}"
                 # Don't do a JSON dump here - too much changes without human
                 # action.
-                mkdir -p "${PROJECT}/services/container"
-                gcloud container clusters list > "${PROJECT}/services/container/clusters.txt"
+                gcloud \
+                    container clusters list \
+                    --format="value(name, location, locations, currentNodeCount, status)" \
```

Comment:
Do we want `currentNodeCount` here?

Reply:
I added it on purpose, but we may want to change it if it becomes a problem.

Reply:
Agree with merging - easy enough to remove it should we choose to, as you say.